cmd/ndt7-client/main.go

```diff
@@ -97,11 +99,12 @@ var (
 		Options: []string{"wss", "ws"},
 		Value:   "wss",
 	}
-	flagBatch    = flag.Bool("batch", false, "emit JSON events on stdout")
+	flagJSON     = flag.Bool("json", false, "emit JSON events on stdout")
 	flagNoVerify = flag.Bool("no-verify", false, "skip TLS certificate verification")
 	flagHostname = flag.String("hostname", "", "optional ndt7 server hostname")
 	flagTimeout  = flag.Duration(
 		"timeout", defaultTimeout, "time after which the test is aborted")
+	flagQuiet = flag.Bool("quiet", false, "emit summary and errors only")
```

Comment:
I quite liked the suggestion by @pboothe to have `-format=json|human|brief` because it seems to me this kinda solves the annoying issue we had with composing emitters in libndt. I am also not so happy with breaking existing scripts that may exist, but, after discussing this with @robertodauria, I understand the potential of this `-format=` design and I am fully on board regarding it. However, please, @robertodauria:

1. if you don't change to `-format=json`, there's a comment above mentioning it, can you fix the comment?

2. if you change the command line arguments, can you also change the docstring and, if needed, the README?

Reply:
Following a suggestion you gave me in a private chat, I've kept `-batch` as an alias to `-format=json` but marked it as deprecated, so it shouldn't disrupt anyone's workflow. Also, I've updated the documentation to reflect these changes.

I would argue, however, that it's hard to provide any kind of API stability guarantee when the client is relatively new, and people directly using `master` should expect breaking changes, especially at this stage.

Using semantic versioning + advising users to depend on tagged releases would make this a non-issue. Maybe it's something it would be good to discuss with the team?

Reply:
Before providing a more comprehensive response, let me say that the issue we're dealing with here is mostly theoretical, since just me and @mattmathis probably use this client.

Speaking from a theoretical point of view, there is clearly a tension here between moving quickly and breaking things that people may be using. When there is a single repository for all the source code, you can move more quickly because you know that you are not breaking expectations. When there are multiple repositories this is more complex and I believe, from the theoretical point of view, that deleting immediately old code is bad.

Speaking of good practices here, I believe we should consider what the [abseil.io team do in terms of compatibility](https://abseil.io/about/compatibility): "We will support language versions, compilers, platforms, and workarounds as needed for 5 years after their replacement is available, when possible". My suggestion in this regard would be to more modestly strive for keeping the previous public interface for three months. In INTP fashion, of course, here I can also argue that three months equal zero months, because people do not read deprecation notices. Trying to find a synthesis between those two points of view, I'd say we either remove the old interface immediately, or we make it such that people notices the deprecation warning.

(You may also notice that my response here is much more comprehensive than the one I gave you in private as I am making an effort towards not using my [system 1 brain](https://en.wikipedia.org/wiki/Thinking,_Fast_and_Slow) when providing feedback. I think specifically the "three months equal zero months" argument is system 2 thinking.)

[...]

> Using semantic versioning + advising users to depend on tagged releases would make this a non-issue.

I disagree that it is a non-issue. Assuming semantic versioning is done correctly, a semantic bump only tells you that you are going to have some kind of trouble. (In this regard, I believe API stability is impossible, and for this reason it probably makes sense to never provide strong API stability, i.e., never release v.1.0.0.)

> Maybe it's something it would be good to discuss with the team?

This discussion is dangerous material, because certainly it has the potential to go down to a rabbit hole. At the same time, it's also a topic that has to do with engineering, and I may end up learning something, so, yes please!